runtime/src/bank/partitioned_epoch_rewards/mod.rs

```diff
@@ -44,18 +45,44 @@ type PartitionedStakeRewards = Vec<PartitionedStakeReward>;
 pub(crate) struct StartBlockHeightAndRewards {
     /// the block height of the slot at which rewards distribution began
     pub(crate) distribution_starting_block_height: u64,
-    /// calculated epoch rewards pending distribution, outer Vec is by partition (one partition per block)
+    /// calculated epoch rewards before partitioning
+    pub(crate) all_stake_rewards: Arc<Vec<PartitionedStakeReward>>,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub(crate) struct StartBlockHeightAndPartitionedRewards {
+    /// the block height of the slot at which rewards distribution began
+    pub(crate) distribution_starting_block_height: u64,
+    /// calculated epoch rewards pending distribution after partitioning, outer Vec is by partition (one partition per block)
     pub(crate) stake_rewards_by_partition: Arc<Vec<PartitionedStakeRewards>>,
 }
 
+impl StartBlockHeightAndRewards {
+    pub(crate) fn do_partition(
```

Comment:
nit: I prefer this code to be inline in the distribution module because it's small and not reused anywhere else.